BranchMetrics/ios-branch-deep-linking-attribution
# iOS Integration Validator Bugfix (for client<>server URI scheme check)

## Reference
No ticket - this was confusing clients, so I wanted to fix this.

## Summary
Currently, our iOS integration validator would check for the Branch URI scheme as the first URI scheme in the list (apart from a few like facebook, etc... that are hardcoded for it to skip when checking). The new code fixes this issue by looping through each of the URI schemes so that Branch doesn't have to be the first URI scheme in the list for the integration validator to pass.

## Motivation
This was confusing clients. 

To see an example of this issue, please view the Master ProServe iOS Branch `example-of-uri-scheme-bug` (Link: https://github.com/BranchMetrics/Master-Proserve-iOS/tree/example-of-uri-scheme-bug)

I added `psmaster0`, an incorrect URI scheme, at index 0. 

![Screenshot 2023-10-23 at 10 10 46 AM](https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/assets/87081315/8b5e7dc8-3a89-4859-9efd-694a5ec2f7b6)

This caused the integration validator check for the URI scheme to fail, even though the correct URI scheme of `psmaster` was in the list right below it. 

<img width="442" alt="Screenshot 2023-10-20 at 4 21 03 PM" src="https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/assets/87081315/ed2e1681-4c82-44ab-9b78-d06c5d60ebf4">

With the changes in this PR, the issue is fixed by checking if the correct URI scheme is present **_anywhere_** in the list

<img width="453" alt="Screenshot 2023-10-23 at 10 04 46 AM" src="https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/assets/87081315/9df16525-c51e-4e32-8d3b-8e8e835e655e">

## Type Of Change


## Testing Instructions
To test this, do one of the following:
- Test the Master ProServe iOS branch I put together where I recreated the bug and then used this sample code to fix it (Link: https://github.com/BranchMetrics/Master-Proserve-iOS/tree/integration-validator-uri-fix-test)
- Utilize your own sample app. Add an incorrect URI scheme (one other than the one you have on your Branch dashboard) as Item 0 in the list. If you run the integration validator, notice the fail message. Then add the code from this PR and observe the issue is fixed.










cc @BranchMetrics/saas-sdk-devs for visibility.
